python/cuml/cuml/manifold/umap.pyx

```diff
@@ -430,6 +436,9 @@ class UMAP(UniversalBase,
 
         self.build_kwds = build_kwds
 
+        if self.build_kwds is not None and any(key.startswith("nnd_") for key in self.build_kwds):
+            raise Exception("build_kwds no longer supports nnd_* arguments. Please refer to docs for detailed configurations.")
```

Comment:
Please use `ValueError` here instead of a generic `Execption` class:
```suggestion
            raise ValueError("build_kwds no longer supports nnd_* arguments. Please refer to docs for detailed configurations.")
```

Reply:
Would it be possible to translate the old `nnd_` pre-fixed arguments into the new format?

Reply:
lines noted as a comment below

Reply:
Yes, but we are introducing breaking API changes in this PR and it seems to me like it would actually be relatively easy to translate the previous intent to the new parameter structure and emit a deprecation warning.